api/include/opentelemetry/context/context.h

```diff
@@ -0,0 +1,244 @@
+      // contains the new key and value data.
+      Context* SetValue(nostd::string_view key, ContextValue value) noexcept
+      {
+        return new Context(key, value, this);
+      }
+
+      // Returns the value associated with the passed in key.
+      common::AttributeValue GetValue(nostd::string_view key)
+      {
+        //Iterate through the context nodes
+        for(Context *context = this; context != nullptr; context = context->next_){
+          //Iterate through the internal data nodes
+          for(DataList *data = context->head_; data != nullptr; data = data->next_){
+            if(strncmp(key.data(), data->key_, data->key_length_) == 0){
+              return data->value_;
+            }
+          }
+        }
+        return "";
+      }
+
+      // Iterates through the current and comparing context objects
+      // to check for equality.
+      bool operator==(const Context &other)
+      {
+        // If the two contexts are not at the place in the list
+        // they cannot be equivalent
+        if(next_ != other.next_){
+          return false; 
+        }
+
+        DataList *other_data = other.head_;
+
+        // Iterate through the lists and compare the keys
+        // TODO should also compare the values but the nostd::variant
+        // seems to not have a comparison operator
+        for(DataList *data = head_; data != nullptr; data = data->next_){
+          if(other_data == nullptr){
+            return false;
+          }
+
+          if((data->key_ != other_data->key_)){
+            return false;
+          }
+
+          data = data->next_;
+          other_data = other_data->next_;
+        }
+
+        return true;
+      }
+      
+      // Copy constructor links to the same next node but copies the other's
+      // data list into a new linked list
+      Context(const Context &other){
+        next_ = other.next_;
+        head_ = new DataList; 
+        DataList *data = head_;
+        // Copy the other data into a new linked list
+        for(DataList* other_data = other.head_; other_data != nullptr; other_data = other_data->next_){
+          if(other_data->next_ != nullptr){
+            data->next_ = new DataList;
+          }
+          else{
+            data->next_ = nullptr ; 
+          }
+          data->key_length_ = other_data->key_length_;
+          data->key_ = new char[data->key_length_];
+          memcpy(data->key_, other_data->key_, data->key_length_);
+          data->value_ = other_data->value_;
+          data = data->next_;
+        }
+      }
+
+      // Assignment constructor links to the same next node but copies the other's
+      // data list into a new linked list
+      Context &operator=(const Context &other) {
+        next_ = other.next_;
+        head_ = new DataList; 
+        DataList *data = head_;
+        // Copy the other data into a new linked list
+        for(DataList* other_data = other.head_; other_data != nullptr; other_data = other_data->next_){
```

Comment:
Be consistent. Here and in rest of the file.
`DataList *something` vs `DataList* something`

https://google.github.io/styleguide/cppguide.html#Pointer_and_Reference_Expressions

Reply:
Changed all to DataList* style.